integration-tests/src/tests/client/state_dump.rs

```diff
@@ -120,3 +133,247 @@ fn test_state_dump() {
         actix_rt::System::current().stop();
     });
 }
+
+fn run_state_sync_with_dumped_parts(
```

Comment:
Please add some high level description of what this does and the reason it was created.

Understanding what and why a test does makes it much easier to fix a test later.